python/arcticdb/util/hypothesis.py

```diff
@@ -65,6 +65,12 @@ def integral_type_strategies(draw):
     return draw(from_dtype(draw(st.one_of([unsigned_integer_dtypes(), integer_dtypes()]))).filter(non_infinite))
 
 
+@st.composite
+def dataframes_with_names_and_dtypes(draw, names, dtype_strategy):
```

Comment:
This is to help with the flaky numeric isin tests. With latest pandas, hypothesis sometimes provides a df containing floats even when using `elements=integral_type_strategies()`. It is safer to explicitly declare the types and leave the elements to be inferred, rather than the other way around.